cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/BasicWorkflowToken.java

```diff
@@ -32,26 +35,31 @@
  */
 public class BasicWorkflowToken implements WorkflowToken {
   private Map<String, Map<String, Long>> mapReduceCounters;
-  private final Map<String, List<NodeValueEntry>> tokenValueMap = Maps.newHashMap();
+  private final Table<Scope, String, List<NodeValueEntry>> tokenValueMap = HashBasedTable.create();
   private String nodeName;
 
   public BasicWorkflowToken() {
   }
 
-  public BasicWorkflowToken(BasicWorkflowToken other) {
-    for (Map.Entry<String, List<NodeValueEntry>> entry : other.tokenValueMap.entrySet()) {
+  public BasicWorkflowToken(Table<Scope, String, List<NodeValueEntry>> tokenValueMap, String nodeName,
+                            @Nullable Map<String, Map<String, Long>> mapReduceCounters) {
+    for (Table.Cell<Scope, String, List<NodeValueEntry>> cell : tokenValueMap.cellSet()) {
```

Comment:
hmmm, there seems to be a [Table#putAll](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/collect/Table.html#putAll(com.google.common.collect.Table) method. Can you use it instead of this loop?